### PR TITLE
Include Secure and Non-Secure TF-M regression tests

### PR DESCRIPTION
**Pull request by andvib**
~~Based on #31910~~

The PR adds the option to run the TF-M regressions tests _directly_ using the Zephyr build system, in the form of a regular Zephyr sample

### CODE REVIEW

File: samples/tfm_integration/tfm_regression_test/src/main.c [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Copyright (c) 2021 Nordic Semiconductor ASA.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+/* Run the PSA test suite */
+void psa_test(void);
+
```
Comment (oyvindronningstad) [inline samples/tfm_integration/tfm_regression_test/src/main.c:11]:
```suggestion
```

File: samples/tfm_integration/tfm_regression_test/src/main.c [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+__attribute__((noreturn))
+void main(void)
+{
+	printk("Should not be printed, ensure TF-M NS application is flashed to the device.\n");
```
Comment (oyvindronningstad) [inline samples/tfm_integration/tfm_regression_test/src/main.c:15]: Wording suggestion. Added panic as well so it fails a bit more loudly.
```suggestion
	printk("Should not be printed, expected TF-M's NS application to be run instead.\n");
	k_panic();
```

File: modules/trusted-firmware-m/Kconfig [hunk 3]
```diff
@@ -229,11 +229,17 @@ config TFM_IPC
...
 	help
 	  When enabled, this option signifies that the TF-M build includes
-	  the Secure and the Non-Secure regression tests.
+	  the Secure domain regression tests.
+
+config TFM_REGRESSION_NS
+	bool "Regression tests Non-Secure"
+	help
+	  When enabled, this option signifies that the TF-M build includes
+	  the Non-Secure domain regression tests.
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:242]: This help text (and maybe the promt also) should make it clear that the Zephyr app is completely replace with TF-M's NS app.

File: samples/tfm_integration/tfm_regression_test/README.rst [hunk 4]
```diff
@@ -0,0 +1,114 @@
+.. _tfm_regression_test:
+
+TF-M Regression Test Sample
+###############################################
+
+Overview
+********
+
+Run the TF-M regression tests using the Zephyr build system.
```
Comment (oyvindronningstad) [inline samples/tfm_integration/tfm_regression_test/README.rst:9]: Mention that this replaces the Zephyr app.

**Review by oyvindronningstad — COMMENTED** [review]
Looks very good! A few, mostly doc-related comments.

File: modules/trusted-firmware-m/Kconfig (hunk 3 above)
Comment (ioannisg) [reply modules/trusted-firmware-m/Kconfig:242]: Yes, I agree about this. We need to stress that @andvib

File: modules/trusted-firmware-m/Kconfig [hunk 5]
```diff
@@ -229,11 +229,17 @@ config TFM_IPC
 	  TF-M Profile option is supplied, do not allow manual setting of
 	  the IPC mode, as it is determined by the profile setting.
 
-config TFM_REGRESSION
-	bool "Regression tests"
+config TFM_REGRESSION_S
+	bool "Regression tests Secure"
```
Comment (ioannisg) [inline modules/trusted-firmware-m/Kconfig:233]: MAybe add the "Secure" and "Non-Secure" in the start of the phrases :)

**Review by ioannisg — COMMENTED** [review]
Thanks for the PR @andvib , some minor comments from my side.
@microbuilder would you like to review this one?

**Comment by microbuilder** [comment]
> Thanks for the PR @andvib , some minor comments from my side.
> @microbuilder would you like to review this one?

Happy to review, yes. I agree it should be clearer that the zephyr app is completely replaced with this sample. The current wording is ambiguous.

File: modules/trusted-firmware-m/Kconfig (hunk 3 above)
Comment (andvib) [reply modules/trusted-firmware-m/Kconfig]: Good point! I tried rephrasing both these and the README for this to be clearer

**Review by ioannisg — APPROVED** [review]

File: modules/trusted-firmware-m/Kconfig [hunk 6]
```diff
@@ -229,11 +229,17 @@ config TFM_IPC
...
+	bool "TF-M Secure Regression tests"
 	help
 	  When enabled, this option signifies that the TF-M build includes
-	  the Secure and the Non-Secure regression tests.
+	  the Secure domain regression tests.
+
+config TFM_REGRESSION_NS
+	bool "Use the TF-M Non-Secure Regression test application"
+	help
+	  When enabled, this option signifies that the application will be
```
Comment (oyvindronningstad) [inline modules/trusted-firmware-m/Kconfig:241]:
```suggestion
	  When this is enabled, the Zephyr application as a whole will be
```

**Review by oyvindronningstad — APPROVED** [review]
LGTM!

**Comment by ioannisg** [comment]
@microbuilder please, take a look :) :)

File: samples/tfm_integration/tfm_regression_test/README.rst [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+Overview
+********
+
+Run both the Secure and Non-secure TF-M Regression tests using the Zephyr build system.
+
+The build system will replace the Zephyr application with the Non-Secure TF-M test application,
+while the Secure tests will be included in the TF-M build itself.
+
+The TF-m regression tests are implemented in the tf-m-tests repo: https://git.trustedfirmware.org/TF-M/tf-m-tests.git/
```
Comment (microbuilder) [inline samples/tfm_integration/tfm_regression_test/README.rst:14]:
```suggestion
The TF-M regression tests are implemented in the tf-m-tests repo: https://git.trustedfirmware.org/TF-M/tf-m-tests.git/
```

File: samples/tfm_integration/tfm_regression_test/README.rst [hunk 8]
```diff
@@ -0,0 +1,117 @@
+.. _tfm_regression_test:
+
+TF-M Regression Test Sample
+###############################################
```
Comment (microbuilder) [inline samples/tfm_integration/tfm_regression_test/README.rst:4]:
```suggestion
TF-M Regression Test Sample
###########################
```

**Review by microbuilder — CHANGES_REQUESTED** [review]
Looks good other than some nit-picky readme changes.

**Review by microbuilder — APPROVED** [review]

**Comment by ioannisg** [comment]
This needs a release notes entry, imo. I'll try to send one later this week.

**Pull request merged**